src/utils/COMETAdapter.cpp

```diff
@@ -0,0 +1,568 @@
+    os << "decoy_search = " << 0 << "\n"; // 0=no (default), 1=concatenated search, 2=separate search
+    os << "num_threads = " << getIntOption_("threads") << "\n";  // 0=poll CPU to set num threads; else specify num threads directly (max 64)
+
+    // masses
+    os << "peptide_mass_tolerance = " << getDoubleOption_("precursor_mass_tolerance") << "\n";
+    os << "peptide_mass_units = " << getIntOption_("precursor_error_units") << "\n";                  // 0=amu, 1=mmu, 2=ppm
+    os << "mass_type_parent = " << 1 << "\n";                    // 0=average masses, 1=monoisotopic masses
+    os << "mass_type_fragment = " << 1 << "\n";                  // 0=average masses, 1=monoisotopic masses
+    os << "precursor_tolerance_type = " << 1 << "\n";            // 0=MH+ (default), 1=precursor m/z; only valid for amu/mmu tolerances
+    os << "isotope_error = " << getIntOption_("isotope_error") << "\n";                      // 0=off, 1=on -1/0/1/2/3 (standard C13 error), 2= -8/-4/0/4/8 (for +4/+8 labeling)
+
+    // search enzyme
+
+    String enzyme_name = getStringOption_("enzyme");
+    String enzyme_number = String(EnzymesDB::getInstance()->getEnzyme(enzyme_name)->getCOMETid());
+
+    os << "search_enzyme_number = " << enzyme_number << "\n";                // choose from list at end of this params file
+    os << "num_enzyme_termini = " << getIntOption_("num_enzyme_termini") << "\n";                  // 1 (semi-digested), 2 (fully digested, default), 8 C-term unspecific , 9 N-term unspecific
+    os << "allowed_missed_cleavage = " << getIntOption_("allowed_missed_cleavages") << "\n";             // maximum value is 5; for enzyme search
+
+    // Up to 9 variable modifications are supported
+    // format:  <mass> <residues> <0=variable/else binary> <max_mods_per_peptide> <term_distance> <n/c-term> <required>
+    //     e.g. 79.966331 STY 0 3 -1 0 0
+    vector<String> variable_modifications_names = getStringList_("variable_modifications");
+    vector<ResidueModification> variable_modifications = getModifications_(variable_modifications_names);
+    if (variable_modifications.size() > 9)
+    {
+      throw OpenMS::Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Error: Comet only supports 9 variable modifications. " + String(variable_modifications.size()) + " provided.");
+    }
+
+    Size var_mod_index = 0;
+
+    // write out user specified modifications
+    for (; var_mod_index < variable_modifications.size(); ++var_mod_index)
+    {
+      const ResidueModification mod = variable_modifications[var_mod_index];
+      double mass = mod.getDiffMonoMass();
+      String residues = mod.getOrigin();  // TODO: check if origin contains C-term string or similar. Should not be passed to commet as residue string
+      String variable = "0";
+      String max_mods_per_peptide = "3";
+      String term_distance = "-1";
+      String nc_term = "0";
+
+      if (residues=="N-term")
```

Comment:
After #2504 is merged, move the "N-term"/"C-term" cases into the handling of the terminal specificities, e.g.:
```
if (mod.getTermSpecificity() == ResidueModification::C_TERM)
{
  if (residue == 'X') residue = 'c';
  ...
}
```
Depending on if/how Comet handles terminal modifications with residue specificity (e.g. "Gln->pyro-Glu (N-term Q)"), you might have to remove the `if (residue == 'X')` check.